gb-emu-lib/CPU.cpp

```diff
@@ -942,65 +944,54 @@ ushort CPU::ReadUShortPC()
 
 void CPU::HandleInterrupts()
 {
-
     // If the IME is enabled, some interrupts are enabled in IE, and
     // an interrupt flag is set, handle the interrupt.
-    if (m_IME)
+    if (m_IME == 0x01)
     {
         byte IE = m_MMU->ReadByte(0xFFFF);
         byte IF = m_MMU->ReadByte(0xFF0F);
-        m_IME = 0x00; // Disable further interrupts
 
-        // Jump to the correct handler
-        if ((ISBITSET(IE, 0)) && (ISBITSET(IF, 0)))
+        // This will only match valid interrupts
+        byte activeInterrupts = ((IE & IF) & 0x0F);
```

Comment:
Magic! If you AND these two, and then AND with 0x0F. After this is done, activeInterrupts will only have bits that give us some action. It'll be 0x00 if there are no matches.